overlays/nodejs/serverless/function.ts

```diff
@@ -41,61 +42,58 @@ export interface Context {
 export type Handler = (event: any, context: Context, callback: (error: any, result: any) => void) => any;
 
 /**
- * FunctionOptions provides configuration options for the serverless Function.
+ * FunctionOptions provides configuration options for the serverless Function.  It is effectively
+ * equivalent to [aws.lambda.FunctionArgs] except with a few important differences documented at the
+ * property level.  For example, [role] is an actual iam.Role instance, and not an ARN. Properties
+ * like [runtime] are now optional.  And some properties (like [code]) are entirely disallowed.
  */
-export interface FunctionOptions {
+export type FunctionOptions = Overwrite<lambda.FunctionArgs, {
     /**
-     * A list of IAM policy ARNs to attach to the Function.  Must provide either [policies] or [role].
+     * Not allowed when creating an aws.serverless.Function.  The [code] will be generated from the
+     * passed in JavaScript callback.
      */
-    policies?: ARN[];
+    code?: never;
+
     /**
-     * A pre-created role to use for the Function.  Must provide either [policies] or [role].
+     * Not allowed when creating an aws.serverless.Function.  The [code] will be generated from the
+     * passed in JavaScript callback.
      */
-    role?: Role;
+    handler?: never;
+
     /**
-     * A timout, in seconds, to apply to the Function.
+     * A list of IAM policy ARNs to attach to the Function.  Must provide either [policies] or [role].
      */
-    timeout?: number;
+    policies?: ARN[];
+
     /**
-     * The memory size limit to use for execution of the Function.
+     * A pre-created role to use for the Function.  Must provide either [policies] or [role].
      */
-    memorySize?: number;
+    role?: Role;
```

Comment:
FWIW, I think we want to permit an `ARN` here too, e.g. `ARN | Role`.

We have started to change all our overlays because if we insist on a strongly typed instance, it can be more difficult to pass in an existing `Role` object that was provisioned outside of Pulumi. (Although you can of course use `Role.get` to fetch it... which is kind of nice due to the stronger typing.)

Reply:
Seems reasonable.  Though i'd prefer to make that change separate to this one :)